clients/web/src/lib/sources/file-entries.ts

```diff
@@ -4,12 +4,14 @@ import { Entry } from "~/lib/proto/sources.ts";
 import { RpcOutputStream } from "@protobuf-ts/runtime-rpc";
 import { Chunk } from "~/lib/proto/sources.ts";
 
+// In the following we "invented" the escape code %DDD to properly escape dots
```

Comment:
If I may suggest, this comment doesn't explain **why**, and there's a realistic chance we forget why you did this in the future. So:

```suggestion
/**
 * Browsers and Routers decode URLs. Since a filename is a piece of a URL, this is breaking our decoding.
 * So we come up with a stamp (`%DDD`) to allow our sanitization to work without getting confused
 **/
```

something like this perhaps ?